buildlog/buildlog_test.go

```diff
@@ -0,0 +1,172 @@
+package buildlog
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type writeCloser struct {
+	io.Writer
+}
+
+func (wc writeCloser) Close() error {
+	return nil
+}
+
+func write(tb testing.TB, filepath string, inputs []string) {
+	tb.Helper()
+
+	for _, input := range inputs {
+		file, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+		if err != nil {
+			tb.Fatal(err)
+		}
+
+		_, err = file.Write([]byte(input))
+		if err != nil {
+			tb.Fatal(err)
+		}
+
+		err = file.Close()
+		if err != nil {
+			tb.Fatal(err)
+		}
+	}
+}
+
+func TestRun(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		tmpFile, err := ioutil.TempFile("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer tmpFile.Close()
+
+		testInputs := []string{
+			`{"t": 1580198209, "m": "test 1", "n": 0, "s": "main"}` + "\n",
+			`{"t": 1580198222, "m": "test 2", "n": 1, "s": "main"}` + "\n",
+		}
+		go write(t, tmpFile.Name(), testInputs)
+
+		parent, cancel := context.WithCancel(context.Background())
+		writer := bytes.NewBuffer(nil)
+		l := log{
+			file:   tmpFile,
+			writer: writer,
+			ctx:    parent,
+			cancel: cancel,
+		}
+
+		go l.Run()
+
+		time.Sleep(3 * time.Second)
```

Comment:
Do we need to wait 3 seconds?  I tried a second or 500 milliseconds and both is ok.
And this magic number appears 3 times so we should make this wait time a variable. 

note: any `sleep` cause running time of tests longer so we should avoid using it.

Reply:
I fixed it.